src/reputation/tasks.py

```diff
@@ -103,21 +108,63 @@ def create_author_contribution(contribution_type, user_id, unified_doc_id, objec
     Contribution.objects.bulk_create(contributions)
 
 
+def get_transaction_receipt(transaction_hash):
+    # Check if connected successfully
+    if not w3.is_connected():
+        raise Exception("Failed to connect to Ethereum node.")
+
+    # Get transaction receipt
+    return w3.eth.get_transaction_receipt(transaction_hash)
+
+
 def check_transaction_success(transaction_hash):
+    return get_transaction_receipt(transaction_hash)["status"] == 1
+
+
+def get_transaction(transaction_hash):
     # Check if connected successfully
     if not w3.is_connected():
-        return "Failed to connect to Ethereum node."
+        raise Exception("Failed to connect to Ethereum node.")
 
-    try:
-        # Get transaction receipt
-        tx_receipt = w3.eth.get_transaction_receipt(transaction_hash)
+    return w3.eth.get_transaction(transaction_hash)
 
-        # Check if transaction was successful
-        return tx_receipt.status == 1
-    except Exception as e:
-        log_error(e)
-        print(e)
-        return False
+
+def get_block(block_number):
+    # Check if connected successfully
+    if not w3.is_connected():
+        raise Exception("Failed to connect to Ethereum node.")
+
+    return w3.eth.get_block(block_number)
```

Comment:
Should we check for a scenario where block number does not exist? I'm not sure if it is possible. Not sure what  would happen in such scenario

Reply:
I don't think this is possible since we check the transaction receipt was successful which I believe would require a block number.